### ISSUE TITLE
SIGSEGV invoking "jx promote"

### ISSUE THREAD

**Issue by cyrille-leclerc** [jenkins-x/jx#357]
```
$ jx promote

Promoting latest version of app my-spring-boot-app-20180306 to namespace jx-staging
WARNING: Failed to try detect the current Jenkins pipeline for cyrille-leclerc/my-spring-boot-app-20180306/add-home-page due to services "jenkins" not found
WARNING: No $JOB_NAME environment variable found so cannot record promotion activities into the PipelineActivity resources in kubernetes
Using pipeline:  build: #
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x1faa9df]

goroutine 1 [running]:
github.com/jenkins-x/jx/pkg/jx/cmd.(*PromoteOptions).createPromoteKey(0xc4202a0420, 0x0, 0x8)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/promote.go:926 +0x5df
github.com/jenkins-x/jx/pkg/jx/cmd.(*PromoteOptions).Promote(0xc4202a0420, 0xc4203a3770, 0xa, 0x0, 0x1, 0xc42039c05c, 0x1b, 0xc4204b78f0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/promote.go:296 +0x425
github.com/jenkins-x/jx/pkg/jx/cmd.(*PromoteOptions).Run(0xc4202a0420, 0xc420365c50, 0x0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/promote.go:198 +0x236
github.com/jenkins-x/jx/pkg/jx/cmd.NewCmdPromote.func1(0xc42037c500, 0x2e868b0, 0x0, 0x0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/pkg/jx/cmd/promote.go:112 +0x75
github.com/jenkins-x/jx/vendor/github.com/spf13/cobra.(*Command).execute(0xc42037c500, 0x2e868b0, 0x0, 0x0, 0xc42037c500, 0x2e868b0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/vendor/github.com/spf13/cobra/command.go:757 +0x2c1
github.com/jenkins-x/jx/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc4202f4280, 0xc4202f5180, 0xc4200b1f01, 0xc4202f4f00)
	/home/jenkins/go/src/github.com/jenkins-x/jx/vendor/github.com/spf13/cobra/command.go:843 +0x334
github.com/jenkins-x/jx/vendor/github.com/spf13/cobra.(*Command).Execute(0xc4202f4280, 0xc4202996ee, 0x2dc6dc0)
	/home/jenkins/go/src/github.com/jenkins-x/jx/vendor/github.com/spf13/cobra/command.go:791 +0x2b
github.com/jenkins-x/jx/cmd/jx/app.Run(0xc4200b1f70, 0x1fd4479)
	/home/jenkins/go/src/github.com/jenkins-x/jx/cmd/jx/app/jx.go:18 +0x90
main.main()
	/home/jenkins/go/src/github.com/jenkins-x/jx/cmd/jx/jx.go:10 +0x22
```

```
$ jx version
WARNING: Failed to find draft version: exit status 1
WARNING: Client: &version.Version{SemVer:"v0.9.0", GitCommit:"6a6b387d84cc7530b45bfff704bd1d71e36082ed", GitTreeState:"clean"}
2018/03/06 14:09:31 grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp: missing address"; Reconnecting to { <nil>}
Error: cannot connect to draftd
NAME        VERSION
jx          1.0.118
Jenkins X   0.0.252
Kubernetes  v1.7.12-gke.1
Helm Client v2.8.1+g6af75a8
Helm Server v2.8.1+g6af75a8
Git         git version 2.15.1

$ jx environment -b
Using environment 'staging' in team 'jx' on server 'https://35.196.213.210'
```

**Comment by cyrille-leclerc** [comment]
when using `jx promote` properly, it worked

**Comment by carlossg** [comment]
Fixed in #570

**Comment by jtnord** [comment]
closing as fixed as per above comment
